tests/cli/test_mila_init_ssh_config.py

```diff
@@ -0,0 +1,349 @@
+    return "\n".join(textwrap.dedent(block) for block in blocks).format(user=user)
+
+
+def _yn(accept: bool):
+    return "y" if accept else "n"
+
+
+def test_creates_ssh_config_file(tmp_path: Path):
+    ssh_config_path = tmp_path / "ssh_config"
+    with set_user_inputs(["y", "bob\r", "y", "y", "y", "y", "y"]) as input_pipe:
+        setup_ssh_config_interactive(tmp_path / "ssh_config", _input=input_pipe)
+    assert ssh_config_path.exists()
+
+
+@pytest.mark.parametrize(
+    "confirm_changes",
+    [False, True],
+)
+@pytest.mark.parametrize(
+    "accept_mila, accept_mila_cpu, accept_mila_gpu, accept_mila_computenode",
+    list(itertools.product([False, True], repeat=4)),
+)
+@pytest.mark.parametrize(
+    "initial_contents",
+    [
+        "",
+        """\
+        # A comment in the file.
+        """,
+        # NOTE: Not using this test case because `ssh_config` will generate the outputs with the
+        # same indent, which will make our `expected_blocks` not match the actual output.
+        # """\
+        # # a comment
+        # Host foo
+        #     HostName foobar.com
+        # """,
+        """\
+        # a comment
+        Host foo
+          HostName foobar.com
+
+        # another comment
+        """,
+    ],
+)
+def test_mila_init_no_existing_entries(
+    initial_contents: str | None,
+    accept_mila: bool,
+    accept_mila_cpu: bool,
+    accept_mila_gpu: bool,
+    accept_mila_computenode: bool,
+    confirm_changes: bool,
+    tmp_path: Path,
+):
+    """Checks what entries are added to the ssh config file when running the corresponding portion
+    of `mila init`.
+    """
+    # TODO: This doesn't completely work with the `questionary` package yet.
+    expected_blocks = []
+    expected_blocks += [expected_block_mila] if accept_mila else []
+    expected_blocks += [expected_block_mila_cpu] if accept_mila_cpu else []
+    expected_blocks += [expected_block_mila_gpu] if accept_mila_gpu else []
+    expected_blocks += [expected_block_compute_node] if accept_mila_computenode else []
+
+    ssh_config_path = tmp_path / ".ssh" / "config"
+    ssh_config_path.parent.mkdir(parents=True, exist_ok=False)
+
+    if initial_contents:
+        initial_contents = textwrap.dedent(initial_contents)
+
+    if initial_contents is not None:
+        with open(ssh_config_path, "w") as f:
+            f.write(initial_contents)
+
+    user_inputs = [
+        "bob\r",  # username
+        _yn(accept_mila),
+        _yn(accept_mila_cpu),
+        _yn(accept_mila_gpu),
+        _yn(accept_mila_computenode),
+        _yn(confirm_changes),
+    ]
+    if not confirm_changes:
+        expected_contents = initial_contents
+    elif initial_contents:
+        # If there is already something in the config file, the new entries should be appended at
+        # the end.
+        # TODO: Confirm this with @Olexa
+        expected_contents = initial_contents + _join_blocks(*expected_blocks)
```

Comment:
ChatGPT tells me that the first matching entry is used so it depends on the behaviour we look for. If we don't want to mess with existing things, then appending will not break anything. But if we do think we know better, then entries should be added first so they take precedence on what was already existing